azkaban-exec-server/src/main/java/azkaban/container/ContainerServlet.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.container;
+
+import azkaban.Constants;
+import azkaban.executor.ConnectorParams;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.utils.FileIOUtils;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.log4j.Logger;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import static azkaban.server.HttpRequestUtils.*;
+import static azkaban.common.ServletUtils.*;
+
+
+public class ContainerServlet extends HttpServlet implements ConnectorParams {
+
+  public static final String JSON_MIME_TYPE = "application/json";
+  private static final Logger logger = Logger.getLogger(ContainerServlet.class);
+  private FlowContainer application;
+
+  public ContainerServlet() { super(); }
+
+  @Override
+  public void init(final ServletConfig config) {
+    this.application =
+        (FlowContainer) config.getServletContext().getAttribute(
+            Constants.AZKABAN_SERVLET_CONTEXT_KEY);
+
+    if (this.application == null) {
+      throw new IllegalStateException(
+          "No flow container defined in this servlet context!");
+    }
+  }
+
+  @Override
+  public void doGet(final HttpServletRequest req, final HttpServletResponse resp)
+      throws ServletException, IOException {
+    handleRequest(req, resp);
```

Comment:
Here, session id is not validated for any GET or POST call. How will the call be authenticated?

Reply:
This should be taken up separately, I think. This is the same as existing ExecutorServlet